### ISSUE TITLE
Move `$query` outside routes?

### ISSUE THREAD

**Issue by Kelin2025** [atomic-router/atomic-router#20]
## Tl;dr
For now each route has `route.$query`. But it's the same for each route.  
That might also be confusing with nested routes (see https://github.com/atomic-router/atomic-router/issues/19 for more details)
## Example
Imagine that we have nested routes
```ts
/* /posts/:postId */
const postRoute = createRoute<{ postId: string }>()
/* /posts/:postId/comments/:commentId */
const postCommentRoute = postRoute.createRoute<{ commentId: string }>()
```
Problem is simple:

If we open `/posts/1/comments/1` and go to `/posts/1/comments/2`, we'll update only `postCommentsRoute`.  
Because only `postCommentRoute` params got changed.  

But if we go `/posts/1/comments/1` to `/posts/1/comments/1?foo=bar`, we'll update both.  
Because `$query` exists in both routes.   

If we do something like `chainRoute({ route: postRoute, effect: getPostFx })`, it will be uselessly triggered by this change.
## Solution
I think it's better to make `$query` external:
```ts
// @/shared/query
import { createQuery } from 'atomic-router'

export const $query = createQuery()

// @/pages/post-comments
import { $query } from '@/shared/query'

// Fetch something based on query updates
sample({
  source: $query,
  filter: postCommentsRoute.$isOpened,
  target: getCommentsFx
})

// @/app/routing
import { $query } from '@/shared/query'

const router = createHistoryRouter({
  routes: [...],
  query: $query  // Router will sync query updates with this store
})
```
## Bonus
Also, it solves the problem of query-syncing
You can just update `$query` directly in order to update query params, and subscribe to `$query` in order to get its updates.  
For example, search request based on query params can be implemented very easily:
```ts
import { $query } from '@/shared/query'

$query
  .on(minPriceChanged, (query, price) => ({ ...query, minPrice }))

sample({
  clock: $query,
  filter: searchGoodsRoute.$isOpened,
  target: searchGoodsFx
})
```

**Comment by sergeysova** [comment]
Looks awesome!

**Comment by sergeysova** [comment]
But I think there should be a local `route.$query` for each `createRoute` instance.

If `route` is opened, `route.$query` should update in parallel with the global `$query`.

If user updates `route.$query` by himself, real query params must be udpated only if `route` is opened. Otherwise there should be no effect.

**Comment by Kelin2025** [comment]
Released in [v0.7.1](https://github.com/atomic-router/atomic-router/releases/tag/v0.7.1)
